File: lab_2/random.java
import java.util.Random;

/**
 *class generates a pseudorandom binary sequence of 128 bit
 */
public class RandomGenerator {

    /**
    * generates and prints to the console a 128-bit binary sequence
    *
    * <p>uses {@link java.util.Random} to generate random numbers 0 or 1</p>
    */
    public static void generate_128() {
        Random rand = new Random();
        StringBuilder sequence = new StringBuilder();

        for (int i = 0; i < 128; ++i) {
            sequence.append(rand.nextInt(2));
        }

        System.out.println("generated seq:");
        System.out.println(sequence.toString());
    }

    public static void main(String[] args) {
        generate_128();
    }
}

File: lab_2/random.cpp
#include <fstream>
#include <iostream>
#include <random>
#include <string>

/**
 * generates a pseudorandom binary sequence of 128 bit
 *
 *uses std::random_device to initialize the random number generator
 *
 *@return std::string string of 128 characters '0' and '1'
 */
std::string generate_128() {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<size_t> number(0, 1);

    std::string seq;
    for (size_t i = 0; i < 128; ++i) {
        seq += std::to_string(number(gen));
    }
    return seq;
}

/**
 * saves a string to the specified file
 *
 * @param dir path to the file to save
 * @param seq the string to save
 */
void save_string(std::string dir, std::string seq) {
    try {
        std::ofstream out;
        out.open(dir);
        if (!out.is_open()) {
            throw std::runtime_error("failed to open file " + dir);
        }
        out << seq << std::endl;
        out.close();
    } catch (const std::exception& e) {
        std::cerr << "unknown error " << e.what() << std::endl;
        throw;
    }
}

int main() {
    std::string seq = generate_128();
    save_string("sequence_cpp.txt", seq);
    std::cout << seq;
}